src/core/layout/qgslayouttable.h

```diff
@@ -54,7 +54,15 @@ typedef QVector< QVector< QVariant > > QgsLayoutTableContents;
  * List of column definitions for a QgsLayoutTable
  * \since QGIS 3.0
 */
-typedef QVector< QgsLayoutTableColumn * > QgsLayoutTableColumns;
+typedef QVector<QgsLayoutTableColumn> QgsLayoutTableColumns;
+
+/**
+ * \ingroup core
+ * List of column definitions for sorting a QgsLayoutTable
+ * \since QGIS 3.14
+*/
+typedef QVector<QgsLayoutTableColumn> QgsLayoutTableSortColumns;
```

Comment:
Reusing QgsLayoutTableColumn seems wasteful here -- that class stores a lot more than the sort related attributes. What do you think about splitting the sort related attributes from QgsLayoutTableColumn into a new dedicated class?

Reply:
That's what I did on first run.
But it brings some nice advantage for the code
* that we can have a single model class instead of two
* less deprecation of methods (1), no need to deprecate the use of the class for another one in certain circumstances
* no need to cast in backward project compatibility (and API if we handle this in `setColumns`)

On the other hand we're talking about 3 enums and 1 double values used for nothing for a couple of columns.

I'd vote to stick as is (my agenda votes for it too ;) )

Reply:
Ok, proceed....!